.github/workflows/build-firmware.yaml

```
@@ -27,99 +27,86 @@ concurrency:
       max-parallel: 5
       fail-fast: false
-      matrix: ${{fromJson(needs.prepare.outputs.matrix)}}
+      matrix:
+        device: [esp32, esp8266]
     steps:
       - name: ⤵️ Check out code from GitHub
         uses: actions/checkout@v6.0.1
-      - name: 🔍 Determine ref - ESPHome packages
+      - name: 🔍 Update ESPHome package ref for dev/PR builds
         if: ${{ !contains(fromJSON('["release", "workflow_call"]'), github.event_name) }}
         run: |
```

actionlint (1.7.10)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/reference/security/secure-use#good-practices-for-mitigating-script-injection-attacks for more details

(expression)